tripy/tripy/frontend/shape.py

```diff
@@ -155,25 +155,30 @@ def __str__(self) -> str:
 
     # addition for shapes is concatenation, not tensor addition
 
+    def _validate_add_argument(self, other):
+        if isinstance(other, Shape):
+            return
+        if not isinstance(other, Sequence) or (len(other) != 0 and not isinstance(other[0], int)):
+            raise_error(
+                "Invalid types for addition with a Tripy Shape: "
+                "Implicit conversions are done only for sequences of Python ints. "
+                "Consider calling tp.Shape for an explicit conversion.",
+                details=[other],
```

Comment:
We can probably move some of this to details. Also we should indicate what `other` is prior to printing it:
```suggestion
                "Invalid type for addition with a Tripy Shape: ",
                details=[
                    "Implicit conversions are done only for sequences of Python ints. ", 
                    "Consider calling tp.Shape for an explicit conversion.",
                    "Note: Argument was: ",
                    other
                ],
```

Reply:
Done as suggested, though I had to update the tests too.